### PR TITLE
Restored FRLG items

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## **Discord contact info**
AsparagusEduardo#6051

### CODE REVIEW

File: src/data/graphics/items.h [hunk 1]
```diff
@@ -2049,7 +2039,11 @@ const u32 gItemIconPalette_RainbowPass[] = INCBIN_U32("graphics/items/icon_palet
...
-const u32 gItemIcon_Gem[] = INCBIN_U32("graphics/items/icons/gem.4bpp.lz");
 const u32 gItemIconPalette_Ruby[] = INCBIN_U32("graphics/items/icon_palettes/ruby.gbapal.lz");
+const u32 gItemIcon_Gem[] = INCBIN_U32("graphics/items/icons/gem.4bpp.lz");
+const u32 gItemIconPalette_Sapphire[] = INCBIN_U32("graphics/items/icon_palettes/sapphire.gbapal.lz");
 
-const u32 gItemIconPalette_Sapphire[] = INCBIN_U32("graphics/items/icon_palettes/sapphire.gbapal.lz");*/
+// Remove these
+/*const u32 gItemIcon_ContestPass[] = INCBIN_U32("graphics/items/icons/contest_pass.4bpp.lz");
+const u32 gItemIconPalette_ContestPass[] = INCBIN_U32("graphics/items/icon_palettes/contest_pass.gbapal.lz");
+*/
```
Comment (LOuroboros) [inline src/data/graphics/items.h:2049]: These seem to be included above, so there's no point in keeping them.

**Review by LOuroboros — CHANGES_REQUESTED** [review]

File: src/data/graphics/items.h (hunk 1 above)
Comment (AsparagusEduardo) [reply src/data/graphics/items.h:2049]: Forgot to delete it, one sec ^^;

Comment (AsparagusEduardo) [reply src/data/graphics/items.h]: Done.

File: src/data/graphics/items.h [hunk 2]
```diff
@@ -1981,7 +1993,6 @@ const u32 gItemIconPalette_Scanner[] = INCBIN_U32("graphics/items/icon_palettes/
 const u32 gItemIcon_StorageKey[] = INCBIN_U32("graphics/items/icons/storage_key.4bpp.lz");
 
 const u32 gItemIcon_KeyToRoom1[] = INCBIN_U32("graphics/items/icons/key_to_room_1.4bpp.lz");
-const u32 gItemIconPalette_Key[] = INCBIN_U32("graphics/items/icon_palettes/key.gbapal.lz");
```
Comment (ultima-soul) [inline src/data/graphics/items.h:1984]: This should remain here as the intention is for the first item icon that uses a shared palette has it right underneath its icon declaration.

File: include/graphics.h [hunk 3]
```diff
@@ -4677,6 +4685,31 @@ extern const u32 gItemIcon_Meteorite[];
...
+extern const u32 gItemIcon_SecretKey[];
+extern const u32 gItemIconPalette_SecretKey[];
+extern const u32 gItemIcon_BikeVoucher[];
+extern const u32 gItemIconPalette_BikeVoucher[];
+extern const u32 gItemIcon_GoldTeeth[];
+extern const u32 gItemIconPalette_GoldTeeth[];
+extern const u32 gItemIcon_CardKey[];
+extern const u32 gItemIconPalette_CardKey[];
+extern const u32 gItemIcon_LiftKey[];
+extern const u32 gItemIconPalette_Key[];
```
Comment (ultima-soul) [inline include/graphics.h:4701]: This is a duplicate and should be removed. See other related comment.

File: src/data/graphics/items.h [hunk 4]
```diff
@@ -2049,7 +2039,6 @@ const u32 gItemIconPalette_RainbowPass[] = INCBIN_U32("graphics/items/icon_palet
 const u32 gItemIcon_Tea[] = INCBIN_U32("graphics/items/icons/tea.4bpp.lz");
 const u32 gItemIconPalette_Tea[] = INCBIN_U32("graphics/items/icon_palettes/tea.gbapal.lz");
 
-const u32 gItemIcon_Gem[] = INCBIN_U32("graphics/items/icons/gem.4bpp.lz");
 const u32 gItemIconPalette_Ruby[] = INCBIN_U32("graphics/items/icon_palettes/ruby.gbapal.lz");
-
-const u32 gItemIconPalette_Sapphire[] = INCBIN_U32("graphics/items/icon_palettes/sapphire.gbapal.lz");*/
+const u32 gItemIcon_Gem[] = INCBIN_U32("graphics/items/icons/gem.4bpp.lz");
+const u32 gItemIconPalette_Sapphire[] = INCBIN_U32("graphics/items/icon_palettes/sapphire.gbapal.lz");
```
Comment (ultima-soul) [inline src/data/graphics/items.h:2044]: `gItemIcon_Gem` should be first to match the order in `include/graphics.h`. `gItemIconPalette_Ruby` should follow right after. `gItemIconPalette_Sapphire` should then be on its own with no accompanying icon. See `gItemIcon_StatusHeal` as an example of this organization for items with shared icons, but different palettes.

File: src/data/items.h [hunk 5]
```diff
@@ -8934,6 +8947,44 @@ const struct Item gItems[] =
...
+        .name = _("FAME CHECKER"),
+        .itemId = ITEM_FAME_CHECKER,
+        .price = 0,
+        .description = sFameCheckerDesc,
+        .importance = 1,
+        .unk19 = 1,
+        .pocket = POCKET_KEY_ITEMS,
+        .type = ITEM_USE_BAG_MENU,
+        .fieldUseFunc = ItemUseOutOfBattle_CannotUse,
+    },
```
Comment (ultima-soul) [inline src/data/items.h:8973]: Make sure to restore data that matches the condition before they were removed from IE. I think the only difference might be the capitalization of names, but it's good to double-check.

**Review by ultima-soul — CHANGES_REQUESTED** [review]

**Comment by AsparagusEduardo** [comment]
Ready for re-review.

File: include/graphics.h (hunk 3 above)
Comment (ultima-soul) [reply include/graphics.h:4701]: I think this was missed.

Comment (AsparagusEduardo) [reply include/graphics.h]: You're right. Fixed!

**Review by ultima-soul — APPROVED** [review]

**Pull request merged**